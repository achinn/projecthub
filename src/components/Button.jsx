import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'


export default class Button extends React.PureComponent {
  static propTypes = {
    isActive:  PropTypes.bool,
    children:  PropTypes.node.isRequired,
    className: PropTypes.string,
  }

  static defaultProps = {
    isActive:  false,
    className: '',
  }

  render() {
    const { isActive, children, ...props } = this.props

    return (
      <button
        {...props}
        type="button"
        className={classNames('btn btn-default', { selected: isActive }, props.className)}
      >
        {children}
      </button>
    )
  }
}
