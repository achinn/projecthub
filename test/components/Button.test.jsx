import React from 'react'
import { shallow } from 'enzyme'

import Button from 'components/Button'

describe('Button', () => {
  let props
  let describedComponent

  const button = () => {
    describedComponent = describedComponent || shallow(<Button {...props} />)
    return describedComponent
  }

  beforeEach(() => {
    props = {
      children: 'Foo',
    }
    describedComponent = undefined
  })

  it('displays a <button> element', () => {
    expect(button().find('button')).toHaveLength(1)
    expect(button().props().className).toBe('btn btn-default  ')
    expect(button().text()).toBe('Foo')
    expect(button().html()).toMatchSnapshot()
  })

  it('has a configurable className', () => {
    props.className = 'foo'
    expect(button().props().className).toBe('btn btn-default  foo')
  })

  describe('when isActive', () => {
    beforeEach(() => {
      props.isActive = true
    })

    it('adds a selected class', () => {
      expect(button().props().className).toBe('btn btn-default selected ')
    })
  })
})
