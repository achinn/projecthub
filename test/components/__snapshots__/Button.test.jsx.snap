// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays a <button> element 1`] = `"<button class=\\"btn btn-default  \\">Foo</button>"`;
